src/main/java/org/jboss/tools/intellij/openshift/actions/component/DebugComponentAction.java

```diff
@@ -0,0 +1,120 @@
+        boolean visible = super.isVisible(selected);
+        if (visible) {
+            ComponentNode componentNode = (ComponentNode) selected;
+            Component component = (Component) componentNode.getUserObject();
+            return (isPushed(component) && isDebuggable(componentNode, component));
+        }
+        return false;
+    }
+
+    private boolean isDebuggable(ComponentNode componentNode, Component component) {
+        ApplicationNode applicationNode = (ApplicationNode) componentNode.getParent();
+        try {
+            Odo odo = ((ApplicationsRootNode) componentNode.getRoot()).getOdo();
+            LazyMutableTreeNode projectNode = (LazyMutableTreeNode) applicationNode.getParent();
+            ApplicationsRootNode root = (ApplicationsRootNode) applicationNode.getRoot();
+            ComponentInfo info = odo.getComponentInfo(root.getClient(), projectNode.toString(), applicationNode.toString(), component.getName());
+            //TODO check here is the port is in use, avoiding the error message below when execute the action.
+            return info.getComponentTypeName().equals("java") || info.getComponentTypeName().equals("nodejs");
+        } catch (IOException e) {
+            UIHelper.executeInUI(() -> Messages.showErrorDialog("Error: " + e.getLocalizedMessage(), "Debug"));
+        }
+        return false;
+    }
+
+    private boolean isPushed(Component component) {
+        return component.getState() == ComponentState.PUSHED;
+    }
+
+    @Override
+    public void actionPerformed(AnActionEvent anActionEvent, TreePath path, Object selected, Odo odo) {
+        ComponentNode componentNode = (ComponentNode) selected;
+        Component component = (Component) componentNode.getUserObject();
+        ApplicationNode applicationNode = (ApplicationNode) componentNode.getParent();
+        LazyMutableTreeNode projectNode = (LazyMutableTreeNode) applicationNode.getParent();
+        //TODO check if local port is modified
+        Integer port = new Integer("5858");
+        if (!isAvailable(port)) {
+            UIHelper.executeInUI(() -> Messages.showErrorDialog("Error, cannot run Odo Debug using port " + port + ".\nPort is already in use.", "Odo Debug"));
+            return;
+        }
+        Project project = anActionEvent.getData(CommonDataKeys.PROJECT);
+
+        RunnerAndConfigurationSettings runSettings = RunManager.getInstance(project).createConfiguration(
+                component.getName() + " Remote Debug with odo", RemoteConfigurationType.getInstance().getFactory());
+
+        RemoteConfiguration remoteConfiguration = (RemoteConfiguration) runSettings.getConfiguration();
+        remoteConfiguration.HOST = "localhost";
+        remoteConfiguration.PORT = port.toString();
+
+        ExecHelper.submit(() -> {
+            try {
+                odo.debug(projectNode.toString(), applicationNode.toString(), component.getPath(), component.getName(), port);
+                try {
```

Comment:
i'm struggling again with the delay when odo is started and the debugger. i tried to get the inputstream back and wait for the message but still the debugger cannot conbnect. the only solution i found for now is to wait exactly more than a period of time ( 6s ) . i also came across a similar issue resolved earlier: #33  can this be related ?